components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -1964,13 +2009,50 @@ public String addSubscription(APIIdentifier identifier, String userId, int appli
 
     public void removeSubscription(APIIdentifier identifier, String userId, int applicationId)
             throws APIManagementException {
-        apiMgtDAO.removeSubscription(identifier, applicationId);
+        API api = getAPI(identifier);
+        boolean isTenantFlowStarted = false;
+
+        if (tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+            isTenantFlowStarted = true;
```

Comment:
Tenant flow started after this line. Should be move this flag after tenant flow has started